backend/app/services/knowledge/structured/text_to_sql.py

```
@@ -0,0 +1,336 @@
+        for col in columns:
+            samples = col.get("sample_values", [])
+            if samples:
+                sample_str = ", ".join(str(s)[:50] for s in samples[:3])
+                lines.append(f"- {col['name']}: {sample_str}")
+
+        return "\n".join(lines) if lines else "No samples available"
+
+    async def _call_llm(
+        self,
+        prompt: str,
+        model_config: Optional[Dict[str, Any]] = None,
```

Ruff (0.15.0)

[warning] Unused method argument: `model_config`

(ARG002)